tensorflow_serving/tools/pip_package/setup.py

```diff
@@ -44,7 +45,11 @@
   sys.argv.remove('--project_name')
   sys.argv.pop(project_name_idx)
 
-_TF_REQ = ['tensorflow'+_TF_VERSION_SANITIZED]
+pkg_released = set(["tensorflow", "tensorflow-cpu"])
+pkg_installed = set(map(lambda pkg: pkg.project_name, pkg_resources.working_set))
```

Comment:
for my understanding, is pkg_resources.working_set from the machine where this package is going to be installed OR is this from the machine where the package is being built? we obviously want the former (list packages installed on the target).

Reply:
https://setuptools.readthedocs.io/en/latest/pkg_resources.html#workingset-objects
```
working_set object is initialized from sys.path when pkg_resources is first imported
```

Yes. the former is right.

Reply:
It was my misunderstanding. That is for the latter, not the former one.

I think there is no solution for `conditional requirements`.

Same as `tensorflow` package,  https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/pip_package/setup.py

Creating new package is another solution with `tensorflow-serving-api-cpu`
And then add `tensorflow-cpu` to `_TF_REQ` when package name is `tensorflow-serving-api-cpu`
ex) https://pypi.org/project/tensorflow-serving-api-cpu/